fastval.go

```diff
@@ -302,9 +302,27 @@ func (val FastVal) ToJsonString() (FastVal, error) {
 	return val, errors.New("invalid type coercion")
 }
 
+func (val FastVal) floatToIntOverflows() bool {
+	intVal := val.AsInt()
+	floatVal := val.GetFloat()
+
+	if floatVal >= 0 && intVal < 0 {
```

Comment:
not sure whether these checks are sufficient. In float to integer conversion, when the integral part does not fit in int64, the end behavior is said to be implementation dependent. I don't think we can assume that the sign will always change.

Reply:
OK that's fair. I'll look at another way.

Reply:
There are some concerns with comparisons using math constants (in C/C++ at least). The recommendation was to use the above method.
https://stackoverflow.com/questions/526070/handling-overflow-when-casting-doubles-to-integers-in-c

Golang may be a little different but the recommended method should be safe regardless.